tools/ac_harness/racing_telemetry.py

```
@@ -0,0 +1,171 @@
+    gas, brake = struct.unpack_from("<2f", buf, 4)
+    gear, rpm = struct.unpack_from("<2i", buf, 16)
+    steer, speed = struct.unpack_from("<2f", buf, 24)
+    accg = struct.unpack_from("<3f", buf, 44)  # [lateral, vertical, longitudinal]
+    slip = struct.unpack_from("<4f", buf, 56)
+    return PhysFrame(packet_id, gas, brake, gear, rpm, steer, speed, accg[0], accg[2], slip)
+
+
+def parse_graphics(buf: bytes) -> GfxFrame:
+    """Parse an ``acpmf_graphics`` buffer (pure)."""
+    if len(buf) < 160:
+        raise ValueError(f"graphics buffer too short: {len(buf)} < 160")
```

Ruff (0.15.17)

[warning] Avoid specifying long messages outside the exception class

(TRY003)